src/ms3/bs4_parser.py

```diff
@@ -2087,8 +2088,28 @@ def get_instrument_name(self, staff_name):
         else:
             return fields_data[staff_name]['trackName']
 
-    def set_instrument(self, staff, trackname):
-        pass
+    def set_instrument(self, staff_name, trackname):
+        if staff_name not in self.parsed_parts.staff2part.keys():
+            raise KeyError(f"Don't recognize key '{staff_name}'")
+        existing_value = self.get_instrument_name(staff_name)
+        new_value = str(trackname)
+        if existing_value is not None and existing_value == new_value:
+            self.logger.debug(f"The {staff_name} was already '{existing_value}' and doesn't need changing.")
+            return
+        new_values = self.INSTRUMENT_DEFAULTS[trackname]
```

Comment:
We need to handle the case of an invalid trackname, too. Also, we have to decide if a default part name is accepted, too: Consider, for example, `Violoncello` which is the default instrument name but the default part name is `Cello`. It could be that in the documentation we will give a list of allowed names and for cello, both would be possible. Also, it might be convenient that we define the keys of `INSTRUMENT_DEFAULTS` lowercase, which allows for easier lookup of the user values by making it lowercase, too.

Reply:
Done  in [e0159ed](https://github.com/johentsch/ms3/pull/57/commits/e0159edffc5b03a818264c6600fd2bba1e7520f6) by adding the method enlarge_instrument_defaults_keys that adds more keys to the INSTRUMENT_DEFAULTS dict, also changed tests to check that it works